src/models/assistant-model.ts

```diff
@@ -87,14 +89,59 @@ export const AssistantModel = types
       }
     });
 
+    const fetchAndSendDataContexts = flow(function* () {
+      try {
+        const contexts = yield getListOfDataContexts();
+        const contextsDetails: Record<string, any> = {};
+        for (const ctx of contexts.values) {
+          const { name } = ctx;
+          const ctxDetails = yield getDataContext(name);
+          contextsDetails[name] = ctxDetails.values;
+        }
+        self.transcriptStore.addMessage(DEBUG_SPEAKER, {description: "Data contexts information", content: formatJsonMessage(contextsDetails)});
+        sendCODAPDocumentInfo(`Data contexts: ${JSON.stringify(contextsDetails)}`);
+      } catch (err) {
+        console.error("Failed to get data contexts:", err);
+        self.transcriptStore.addMessage(DEBUG_SPEAKER, {description: "Failed to get data contexts", content: formatJsonMessage(err)});
+      }
+    });
+
+    const sendDataCtxChangeInfo = flow(function* (msg: string) {
+      try {
+        if (self.isLoadingResponse) {
+          self.messageQueue.push(msg);
+        } else {
+          yield sendCODAPDocumentInfo(msg);
+        }
+      } catch (err) {
+        console.error("Failed to send data context info to LLM:", err);
+        self.transcriptStore.addMessage(DEBUG_SPEAKER, {description: "Failed to send data context info to d LLM", content: formatJsonMessage(err)});
+      }
+    });
+
+    const sendCODAPDocumentInfo = flow(function* (message) {
+      try {
+        self.transcriptStore.addMessage(DEBUG_SPEAKER, {description: "Sending CODAP document information to LLM", content: message});
```

Comment:
Should this be using `formatJsonMessage(message)` instead of just `message`?

Reply:
It's a little convoluted but no, because the message that is sent to sendCODAPDocumentInfo is already formatted. It's because this message is a little different since the information is grabbed from CODAP in the listener and then formatted at that point.